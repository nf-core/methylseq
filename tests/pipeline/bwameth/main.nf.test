nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "bwameth"
    tag "pipeline"

    test("Bwameth Single-End") {
        when {
            params {
                aligner        = "bwameth"
                save_reference = true
                outdir         = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("single_end_software_versions") },
                { assert snapshot(path("$outputDir/bwameth/alignments/samtools_stats/Ecoli.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/Ecoli.stats"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222.stats"),
                            path("$outputDir/bwameth/deduplicated/Ecoli.markdup.sorted.bam"),
                            path("$outputDir/bwameth/deduplicated/Ecoli.markdup.sorted.bam.bai"),
                            path("$outputDir/bwameth/deduplicated/SRR389222.markdup.sorted.bam"),
                            path("$outputDir/bwameth/deduplicated/SRR389222.markdup.sorted.bam.bai"),
                            path("$outputDir/bwameth/reference_genome/").list(),
                            path("$outputDir/cat/SRR389222.merged.fastq.gz"),
                            path("$outputDir/methyldackel/Ecoli.markdup.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/SRR389222.markdup.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/mbias/Ecoli.mbias.txt"),
                            path("$outputDir/methyldackel/mbias/SRR389222.mbias.txt"),
                            path("$outputDir/qualimap/Ecoli/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222/genome_results.txt")).match("single_end") },
                { assert new File("$outputDir/fastqc/Ecoli_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/Ecoli_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_fastqc.html").exists() },
                { assert new File("$outputDir/multiqc/bwameth/multiqc_report.html").exists() },
                { assert new File("$outputDir/trimgalore/logs/Ecoli_1.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/logs/Ecoli_2.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/trimgalore/logs/SRR389222.fastq.gz_trimming_report.txt").exists() },
                { assert new File("$outputDir/bwameth/deduplicated/picard_metrics/Ecoli.markdup.sorted.MarkDuplicates.metrics.txt").exists() },
                { assert new File("$outputDir/bwameth/deduplicated/picard_metrics/SRR389222.markdup.sorted.MarkDuplicates.metrics.txt").exists() }
            )
        }
    }

    test("Bwameth Single-End With Index") {
        when {
            params {
                aligner        = "bwameth"
                bwa_meth_index = "https://github.com/nf-core/test-datasets/raw/methylseq/reference/Bwameth_Index.tar.gz"
                methyl_kit     = true
                outdir         = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
            )
        }
    }

    test("Bwameth Single-End RRBS") {
        when {
            params {
                aligner        = "bwameth"
                save_reference = true
                skip_trimming  = true
                rrbs           = true
                save_reference = true
                outdir         = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("single_end_rrbs_software_versions") },
                { assert snapshot(path("$outputDir/bwameth/alignments/samtools_stats/Ecoli.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/Ecoli.stats"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222.flagstat"),
                            path("$outputDir/bwameth/alignments/samtools_stats/SRR389222.stats"),
                            path("$outputDir/bwameth/reference_genome/").list(),
                            path("$outputDir/cat/SRR389222.merged.fastq.gz"),
                            path("$outputDir/methyldackel/Ecoli.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/SRR389222.sorted_CpG.bedGraph"),
                            path("$outputDir/methyldackel/mbias/Ecoli.mbias.txt"),
                            path("$outputDir/methyldackel/mbias/SRR389222.mbias.txt"),
                            path("$outputDir/qualimap/Ecoli/genome_results.txt"),
                            path("$outputDir/qualimap/SRR389222/genome_results.txt")).match("single_end_rrbs") },
                { assert new File("$outputDir/fastqc/Ecoli_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/Ecoli_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/SRR389222_fastqc.html").exists() },
                { assert new File("$outputDir/multiqc/bwameth/multiqc_report.html").exists() }
            )
        }
    }

    test("Bwameth Single End RRBS With Index") {
        when {
            params {
                aligner = "bwameth"
                skip_trimming = true
                rrbs = true
                bwa_meth_index = "https://github.com/nf-core/test-datasets/raw/methylseq/reference/Bwameth_Index.tar.gz"
                methyl_kit = true
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
            )
        }
    }

}
