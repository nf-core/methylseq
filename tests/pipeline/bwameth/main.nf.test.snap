{
    "single_end_software_versions": {
        "content": [
            "{BWAMETH_ALIGN={bwameth=0.2.2}, BWAMETH_INDEX={bwameth=0.2.2}, CAT_FASTQ={cat=8.30}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, FASTQC={fastqc=0.11.9}, QUALIMAP_BAMQC={qualimap=2.2.2-dev}, SAMPLESHEET_CHECK={python=3.8.3}, TRIMGALORE={cutadapt=3.4, trimgalore=0.6.7}, Workflow={nf-core/methylseq=2.4.0}}"
        ],
        "timestamp": "2023-05-28T00:53:43+0000"
    },
    "single_end_rrbs_software_versions": {
        "content": [
            "{BWAMETH_ALIGN={bwameth=0.2.2}, BWAMETH_INDEX={bwameth=0.2.2}, CAT_FASTQ={cat=8.30}, CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, FASTQC={fastqc=0.11.9}, QUALIMAP_BAMQC={qualimap=2.2.2-dev}, SAMPLESHEET_CHECK={python=3.8.3}, Workflow={nf-core/methylseq=2.4.0}}"
        ],
        "timestamp": "2023-05-28T00:53:43+0000"
    },
    "single_end": {
        "content": [
            "Ecoli_10K_methylated.flagstat:md5,f520e87cc4050799a0dbed72909572ec",
            "SRR389222_sub1.flagstat:md5,ad8159a7683b9c3057bf54ccb77a2ddc",
            "SRR389222_sub2.flagstat:md5,2166a52f81f9b3881e8ac060876aa2d9",
            "Ecoli_10K_methylated.markdup.sorted_CpG.bedGraph:md5,d3bce01078b6e7749f3906bf9faad5b5",
            "SRR389222_sub1.markdup.sorted_CpG.bedGraph:md5,0944ad0e3b05791bf6dfd06be7ca4af1",
            "SRR389222_sub2.markdup.sorted_CpG.bedGraph:md5,0a3c19af55d1625e91903ae25d1d2098",
            "genome_results.txt:md5,1ee2e29ac41c2cd7e6a0f00741e8d16e",
            "genome_results.txt:md5,2c755b15ae5ad4a1d29395320a6264ca",
            "genome_results.txt:md5,3c7d14ea070df510941c55f72706a7a0"
        ],
        "timestamp": "2023-05-28T00:53:43+0000"
    },
    "single_end_rrbs": {
        "content": [
            "Ecoli_10K_methylated.flagstat:md5,7c3bae0f9c700dde88785a7e94c6ae2c",
            "SRR389222_sub1.flagstat:md5,897d500a710a56a7098172167fa71108",
            "SRR389222_sub2.flagstat:md5,2d680c5657fc029b703b217be4af68e2",
            "Ecoli_10K_methylated.sorted_CpG.bedGraph:md5,3656c54896013b76de978ab95c3f5975",
            "SRR389222_sub1.sorted_CpG.bedGraph:md5,4475a9462ce86a019015b69b82a9d5a2",
            "SRR389222_sub2.sorted_CpG.bedGraph:md5,18b8845d9d985d3c203ca6c9c40aab1b",
            "genome_results.txt:md5,ca30867e16c5ca25c850c81d88563d41",
            "genome_results.txt:md5,f01c68dd46ca9ec0af3d6ec926763b75",
            "genome_results.txt:md5,f4513f5aa1622937ccdcfe92a6f877c0"
        ],
        "timestamp": "2023-05-28T00:53:43+0000"
    }
}